encoder/ih264e_encode.c

```diff
@@ -109,6 +109,278 @@
 /* Function Definitions                                                      */
 /*****************************************************************************/
 
+#ifdef KEEP_THREADS_ACTIVE
+/**
+*******************************************************************************
+*
+* @brief
+*  Initializes the thread pool for multi-threaded encoding.
+*
+* @par Description:
+*  Creates and initializes the thread pool based on the number of configured
+*  cores. It spawns worker threads and sets up necessary synchronization
+*  mechanisms.
+*
+* @param[in] ps_codec
+*  Pointer to the codec context structure.
+*
+* @returns  IV_SUCCESS on success, IV_FAIL on failure.
+*
+*******************************************************************************
+*/
+WORD32 ih264e_thread_pool_init(codec_t *ps_codec)
+{
+    /* temp var */
+    WORD32 i = 0, j = 0, ret = 0;
+
+    /* thread pool */
+    thread_pool_t *ps_pool = &ps_codec->s_thread_pool;
+
+    /* Return if already initialized */
+    if (ps_pool->i4_init_done == 1) return IV_SUCCESS;
+
+    /* initializing thread pool initialization done */
+    ps_pool->i4_init_done = 0;
+
+    /* initializing end of stream */
+    ps_pool->i4_end_of_stream = 0;
+
+    /* initializing active threads */
+    ps_pool->i4_working_threads = 0;
+
+    /* Initialize new frame ready flag */
+    ps_pool->i4_has_frame = 0;
+
+    /* thread pool mutex init */
+    ret = ithread_mutex_init(ps_codec->s_thread_pool.pv_thread_pool_mutex);
+    if (ret != 0) {
```

Comment:
nit: braces in the next line

please review all and change accordingly

Reply:
similar changes are needed in multiple places. Please review the changes and update.